src/app/components/genre-form-dialog/genre-form-dialog.component.ts

```diff
@@ -15,18 +25,47 @@ import {
 })
 export class GenreFormDialogComponent implements OnInit {
   form: FormGroup;
+  modalTitle: string;
+  submitTitle: string;
+  placeholder: string;
+  name: string;
+  editMode: boolean;
 
   constructor(
     private dialogRef: MatDialogRef<boolean>,
     private genreService: GenreService,
     private fb: FormBuilder,
-    @Inject(MAT_DIALOG_DATA) private data
+    @Inject(MAT_DIALOG_DATA) private data: GenreFormDialogData
   ) {}
 
   ngOnInit(): void {
+    if (this.data.editMode) {
+      this.modalTitle = 'Edit genre';
+      this.submitTitle = 'Update';
+      this.placeholder = 'Genre';
+      this.name = this.data.genre.name;
+      this.editMode = true;
+    } else {
+      this.modalTitle = 'Add new genre';
+      this.submitTitle = 'Save';
+      this.placeholder = 'New genre';
+      this.name = '';
+      this.editMode = false;
+    }
+
     this.form = this.fb.group({
-      name: new FormControl('', Validators.required),
+      name: new FormControl(this.name, [
```

Comment:
Replace `this.name` with `this.data.name ?: ''` and then remove name field from class

Reply:
it does not seem to work correctly with this solution
console throw an error